### ISSUE TITLE
Moving up down is not preserving keyboardDistanceFromTextField

### ISSUE THREAD

**Issue by pgawlowski** [hackiftekhar/IQKeyboardManager#1365]
### Description

Moving up and down through UITextFields is not preserving the distance from textField.

### Steps to Reproduce

1. Let's set 
`        IQKeyboardManager.shared.keyboardDistanceFromTextField = 50
`
2. Select TextField, keyboard appears, distance is correct
3. Click next - distance still corect
4. Click previous - distance is significantly smaller
5. Repeat few times - distance is reduced to 0

**Expected behavior:** 
Well keep the distance.

**Actual behavior:**
Not keeping the distance :(

**Reproduces how often:**
Right now tested for TextFields in StackView emebdded in ScrollView

### Versions

**Xcode:** 9.4

**Mac OS:** 10.13.4

**Simulator/Device:** 11.4

**Simulator/Device Name:** iPhone SE

**Library Version:** 5.0.7

### Additional Information

![jun-13-2018 02-06-47](https://user-images.githubusercontent.com/9845024/41323406-7465c770-6eae-11e8-9ca6-f56890f8ff2a.gif)

**Comment by hackiftekhar** [comment]
@pgawlowski Can you please share the demo project with me since it shouldn't happen.

**Comment by pgawlowski** [comment]
@hackiftekhar here is repository with exactly same form I am using in my project which has the issues. I've included all related files like IBDesignables, Rx handlings etc. 
Use iPhone SE for testing purpose because it has the smallest operational area.

I am adding additional gifs showing what exact problems we're dealing with.

![jupmy](https://user-images.githubusercontent.com/9845024/42117655-29c8f586-7bfe-11e8-9c6a-9747c837fb14.gif)
Let's start with sth I called "jumpy" keyboard. While moving up and down constantly the scroll distance is growing up to infinity. This is probably caused because position is being calculated not as a constant position but current offset on screen.

![issue_2- not sticky keyboard](https://user-images.githubusercontent.com/9845024/42117688-60b7e156-7bfe-11e8-95d3-60e1107f9ccf.gif)
Second gif is showing another issue. We're moving down. And I am expecting that textField after scroll will be positioned directly over keyboard. On this screen offset is too bit and textField is positioned on top of screen.

**Comment by renzifeng** [comment]
I have this problem too

**Comment by hackiftekhar** [comment]
@pgawlowski Thanks for the demo project. I'll see the demo and try to reproduce this.

**Comment by hackiftekhar** [comment]
I checked the demo project and I found that the library is doing correct calculation, but I don't know why scrollView is over-scrolling automatically. For example the library tries to set scrollView.contentOffset as 183 but scrollView scroll till 353 contentOffset. That's why I'm not sure where might be the problem.

**Comment by pgawlowski** [comment]
@hackiftekhar 
My first thought was that this is directly connected to bouncing of scrollView but tried to disable without success. 

Can you point the method you've been investigating. I will try to do some research on my own but because of lack of time I had no chance to in-depth debug of IQ ;)

**Comment by hackiftekhar** [comment]
If you see IQKeyboardManager line 1138
```
scrollView.contentOffset = CGPoint(x: scrollView.contentOffset.x, y: shouldOffsetY)
```
you'll see that this calculate 183 shouldOffsetY, but it internally set 353 and you see textField too above the keyboard.

To confirm this, I added another breakpoint at IQKeyboardManager line 998
```
var superView = textFieldView.superviewOfClassType(UIScrollView.self) as? UIScrollView
```
I tap on previous button and `print superView.contentOffset` and it return 353.

**Comment by pgawlowski** [comment]
@hackiftekhar instead of breakpoint use logging. The problem is not with the initial calculation, breakpoint is affecting the calculations as well, because after breakpoint you're unable to tap previous/next button fast enough so the "bounce" offset is not being included in new calculations. I am nearly 100% sure that this is connected with "bounce" effect.

**Comment by hackiftekhar** [comment]
@pgawlowski If you don't want to use breakpoints for debugging then probably you can also set `IQKeyboardManager.shared.enableDebugging = true` to print logs.

**Comment by pgawlowski** [comment]
@hackiftekhar - any progress? I was also unable to investigate the reason :(

**Comment by hackiftekhar** [comment]
I'm out of country since last 25 days and will be back to home after 20-25 more days. Unable to check.

**Comment by akshit-zaveri** [comment]
Any luck? @hackiftekhar

**Comment by hackiftekhar** [comment]
No luck sir, looms like scrollview internally changing the content offset.

**Comment by hackiftekhar** [comment]
I posted some of my findings on #1541

**Comment by hackiftekhar** [comment]
By comparing at @pgawlowski demo and @lisafly2014 demo (#1541) I found a common thing. Both demo have a UIScrollView->UIStackView->UITextView hierarchy.

On further investigation, I noticed that this issue mostly happens when above kind of hierarchy found, I also found that using `setContentOffset:animated:` method fix the problem (but created problems for other logics).

So as a workaround, if above kind of hierarchy found then Library will use `setContentOffset:animated:` method, otherwise library will use non-animated method like `contentOffset = offset`. Note that the keyboardDistanceFromTextField will not be preserved with this kind of hierarchy but the jumping effect will be reduced.
